src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> element;
+    private int size;
+
+    private class Node<T> {
+        private T item;
+        private Node<T> next;
+
+        public Node(T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+        }
+    }
 
+    public void push(T value) {
+        if (size == 0) {
+            element = new Node<>(value, null);
+        } else {
+            element = new Node<>(value, element);
+        }
+        size++;
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(value, element);
        element = newNode;
        size++;
```
Can we not use (if and else) here ? We can do like this.

Reply:
done